Spigot-API-Patches/0148-Expose-attack-cooldown-methods-for-Player.patch

```diff
@@ -0,0 +1,40 @@
+From b0307c1c84d01e61a96ad64b5153abd236229626 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Tue, 4 Sep 2018 15:01:54 -0500
+Subject: [PATCH] Expose attack cooldown methods for Player
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 6da68835..ae2b9752 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1909,6 +1909,26 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
+      * @param profile The new profile to use
+      */
+     void setPlayerProfile(PlayerProfile profile);
++
++    /**
++     * Returns the amount of ticks the current cooldown lasts
++     *
++     * @return Amount of ticks cooldown will last
++     */
++    float getCooldownPeriod();
++
++    /**
++     * Returns the percentage of attack power available based on the cooldown (zero to one).
++     *
++     * @param adjustTicks Amount of ticks to add to cooldown counter for this calculation
++     * @return Percentage of attack power available
++     */
++    float getCooledAttackStrength(float adjustTicks);
```

Comment:
As I said on the issue comment, I don't think the API should have a `float` param, the caller can add the value themselves. I vote to create an `AttackStrengthType` enum (the name can be changed ofc) which would list the usages by NMS code, as I feel that people trying to use this method would use 0 per the javadoc param comment when they might have intended to use another value.

Reply:
I left it as a float so the plugin author could fine tune it to their own needs outside of anything we could possibly think of ahead of time.

Reply:
Yes, what I meant with that edit was keeping the method signature the same but providing some constants somewhere 😛

Reply:
I honestly wouldn't even know what to call those 3 values if we were to provide then in enum/constants. They seem more like tweaks to me that don't have any real meaning.

Reply:
docs are fine to explain it. let plugins figure out what they want